### PR TITLE
Profile.Advice.json: Expand on $profile paths

### PR DESCRIPTION
**Pull request by Maamue**
# PR Summary

Expand on Profile Advice as asked in #260 

## Context

- expand on $profile possibilities (different PSHosts)
- expand on changes IDEs make on $profile

## Changes

only textual changes with different possibilities where $profile can point to

## Checklist

- [x] Pull Request has a meaningful title.
- [x] Summarised changes.
- [x] Pull Request is ready to merge & is not WIP.
- [x] Added tests / only testable interactively.
  - Make sure you add a new test if old tests do not effectively test the code changed.
- [ ] Added documentation / opened issue to track adding documentation at a later date.

### CODE REVIEW

File: PSKoans/Data/Advice/Customization/Profile.Advice.json [hunk 1]
```diff
@@ -4,6 +4,16 @@
...
-    "  notepad $profile"
+    "  notepad $profile",
+    "\n",
+    "The $profile variable stores the path to the 'Current User, Current Host' profile, that points to a Microsoft.PowerShell_profile.ps1 file. Other profiles can be retreived from note properties of the $profile variable:",
+    "AllUsersAllHosts",
+    "AllUsersCurrentHost",
+    "CurrentUserAllHosts",
+    "CurrentUserCurrentHost",
+    "\n",
+    "Different PSHosts (applications hosting a powershell process) like IDEs have custom profiles paths that are referenced by $profile. While the default $profile will be available in every environment, it is useful to know where IDEs save their profiles. Customizations only required for e.g. development can be added.",
```
Comment (vexx32) [inline PSKoans/Data/Advice/Customization/Profile.Advice.json:15]:
```suggestion
    "Different PSHosts (applications hosting a powershell process) like IDEs have custom profile paths that are referenced by $profile. While the default $profile will be available in every environment, it is useful to know where IDEs save their profiles. Customizations only required for e.g. development can be added.",
```

File: PSKoans/Data/Advice/Customization/Profile.Advice.json [hunk 2]
```diff
@@ -4,6 +4,16 @@
...
+    "Powershell ISE uses $profile, but with a custom path pointing to a Microsoft.PowerShellISE_profile.ps1 file.",
+    "Visual Studio Code uses $profile, but with a custom path pointing to a Microsoft.VSCode_profile.ps1 file."
```
Comment (vexx32) [inline PSKoans/Data/Advice/Customization/Profile.Advice.json:17]: I think we should clarify these custom paths are incorporated into the "current host" variants of the profile paths.

**Review by vexx32 — CHANGES_REQUESTED** [review]
Thanks for taking care of this! 💖 

There's one or two small things that I think we should alter a little but other than that, this looks fantastic. Thanks! 😊 🌸

File: PSKoans/Data/Advice/Customization/Profile.Advice.json (hunk 2 above)
Comment (Maamue) [reply PSKoans/Data/Advice/Customization/Profile.Advice.json]: Good catch! Added another sentence to explain the same behaviour.

File: PSKoans/Data/Advice/Customization/Profile.Advice.json (hunk 1 above)
Comment (Maamue) [reply PSKoans/Data/Advice/Customization/Profile.Advice.json]: should be fixed now, thanks :)

**Review by vexx32 — APPROVED** [review]

**Pull request merged**